metrics/cli/test/gcloud/adapter/gcloud_cli_service_adapter_test.dart

```diff
@@ -59,143 +60,213 @@ void main() {
       },
     );
 
     test(
-      ".createProject() returns the identifier of the created project",
+      ".createProject() doesn't call any other methods if GCloud CLI throws during the project app creation",
       () async {
-        final projectId = await gcloudAdapter.createProject();
+        when(gcloudCli.createProjectApp(any, any))
+            .thenAnswer((_) => Future.error(stateError));
 
-        expect(projectId, isNotNull);
-        verify(gcloudCli.createProject(projectId)).called(once);
+        await expectLater(gcloudService.createProject(), throwsStateError);
+
+        verify(gcloudCli.createProject(any)).called(once);
+        verify(gcloudCli.listRegions(any)).called(once);
+        verify(prompter.prompt(any)).called(once);
+        verify(gcloudCli.createProjectApp(any, any)).called(once);
+
+        verifyNoMoreInteractions(gcloudCli);
+        verifyNoMoreInteractions(prompter);
       },
     );
 
     test(
-      ".version() shows the version information",
+      ".createProject() doesn't call any other methods if GCloud CLI throws during the Firestore API enabling",
       () async {
-        await gcloudAdapter.version();
+        when(gcloudCli.enableFirestoreApi(any))
+            .thenAnswer((_) => Future.error(stateError));
+
+        await expectLater(gcloudService.createProject(), throwsStateError);
+
+        verify(gcloudCli.createProject(any)).called(once);
+        verify(gcloudCli.listRegions(any)).called(once);
+        verify(prompter.prompt(any)).called(once);
+        verify(gcloudCli.createProjectApp(any, any)).called(once);
+        verify(gcloudCli.enableFirestoreApi(any)).called(once);
+
+        verifyNoMoreInteractions(gcloudCli);
+        verifyNoMoreInteractions(prompter);
+      },
+    );
+
+    test(
+      ".createProject() throws if GCloud CLI throws during the database creation",
+      () async {
+        when(gcloudCli.createDatabase(any, any))
+            .thenAnswer((_) => Future.error(stateError));
+
+        await expectLater(gcloudService.createProject(), throwsStateError);
+
+        verify(gcloudCli.createProject(any)).called(once);
+        verify(gcloudCli.listRegions(any)).called(once);
+        verify(prompter.prompt(any)).called(once);
+        verify(gcloudCli.createProjectApp(any, any)).called(once);
+        verify(gcloudCli.enableFirestoreApi(any)).called(once);
+        verify(gcloudCli.createDatabase(any, any)).called(once);
+
+        verifyNoMoreInteractions(gcloudCli);
+        verifyNoMoreInteractions(prompter);
+      },
+    );
+
+    test(
+      ".version() throws if GCloud CLI throws during the version showing",
+      () async {
+        when(gcloudCli.version()).thenAnswer((_) => Future.error(stateError));
+
+        await expectLater(gcloudService.version(), throwsStateError);
 
         verify(gcloudCli.version()).called(once);
+
+        verifyNoMoreInteractions(gcloudCli);
+        verifyNoMoreInteractions(prompter);
```

Comment:
```suggestion
        expect(gcloudService.version(), throwsStateError);
```
I believe in such tests we should just test that the method throws an exception. What do you think?